src/gtc/passes/gtir_pipeline.py

```diff
@@ -0,0 +1,41 @@
+from typing import Callable, Dict, Optional, Sequence, Tuple
+
+from gtc import gtir
+from gtc.passes.gtir_dtype_resolver import resolve_dtype
+from gtc.passes.gtir_prune_unused_parameters import prune_unused_parameters
+from gtc.passes.gtir_upcaster import upcast
+
+
+PASS_T = Callable[[gtir.Stencil], gtir.Stencil]
+
+
+class GtirPipeline:
+    """
+    GTIR passes pipeline runs passes in order and allows skipping.
+
+    May only call existing passes and may not contain any pass logic itself.
+    """
+
+    def __init__(self, node: gtir.Stencil):
+        self.gtir = node
+        self._cache: Dict[Tuple[PASS_T, ...], gtir.Stencil] = {}
```

Comment:
We are not deeply relying on it, if the need arises we can create an enum for passes or something along the lines. For now that seemed like overengineering.

Since the need for skipping is gone for now, we can also drop the whole caching and even any sequences of passes and instead just call the passes in order in the `.full()` method.